lib/newrelic_httprb/instrumentation.rb

```diff
@@ -0,0 +1,33 @@
+DependencyDetection.defer do
+  named :http_rb
+
+  depends_on do
+    defined?(HTTP) && defined?(HTTP::Client)
+  end
+
+  executes do
+    ::NewRelic::Agent.logger.info 'Installing http.rb instrumentation'
+    require 'new_relic/agent/cross_app_tracing'
+    require 'newrelic_http/wrappers'
+  end
+
+  executes do
+    class HTTP::Client
+      def perform_with_newrelic_trace(request, options)
+        wrapped_request = ::NewRelicHTTP::HTTPRequest.new(request)
+
+        ::NewRelic::Agent::CrossAppTracing.tl_trace_http_request(wrapped_request) do
+          # RUBY-1244 Disable further tracing in request to avoid double
+          # counting if connection wasn't started (which calls request again).
```

Comment:
Hmm, you're right. Maybe the solution here would be to fork newrelic, and add the testing code on the fork, and test it as part of their harness. It's not great, but we do get to take advantage of the many testcases they have already written for strange corner cases.